scipy/_lib/doccer.py

```diff
@@ -123,63 +137,112 @@ def _doc(func):
         if cls_docstring is None:
             return func
-        end_of_notes = cls_docstring.find('        References\n')
+        end_of_notes = cls_docstring.find("        References\n")
         if end_of_notes == -1:
-            end_of_notes = cls_docstring.find('        Examples\n')
+            end_of_notes = cls_docstring.find("        Examples\n")
             if end_of_notes == -1:
                 end_of_notes = len(cls_docstring)
-        func.__doc__ = (cls_docstring[:end_of_notes] + notes +
-                        cls_docstring[end_of_notes:])
+        func.__doc__ = (
+            cls_docstring[:end_of_notes] + notes + cls_docstring[end_of_notes:]
+        )
         return func
+
     return _doc
 
 
-def replace_notes_in_docstring(cls, notes):
-    """
-    This decorator replaces the decorated function's docstring
+def replace_notes_in_docstring(cls: type[Any] | Any, notes: str) -> Decorator[R]:
+    """This decorator replaces the decorated function's docstring
     with the docstring from corresponding method in `cls`.
     It replaces the 'Notes' section of that docstring with
     the given `notes`.
+
+    Parameters
+    ----------
+    cls : object
+        A class with a method with the same name as the decorated method.
+        The docstring of the method in this class replaces the docstring of the
+        decorated method.
+    notes : str
+        The notes to replace the existing 'Notes' section with.
+
+    Returns
+    -------
+    decfunc : function
+        The decorator function that modifies the __doc__ attribute
+        of its argument.
     """
-    def _doc(func):
+
+    def _doc(func: Callable[P, R]) -> Callable[P, R]:
         cls_docstring = getattr(cls, func.__name__).__doc__
-        notes_header = '        Notes\n        -----\n'
+        notes_header = "        Notes\n        -----\n"
         # If python is called with -OO option,
         # there is no docstring
         if cls_docstring is None:
             return func
         start_of_notes = cls_docstring.find(notes_header)
-        end_of_notes = cls_docstring.find('        References\n')
+        end_of_notes = cls_docstring.find("        References\n")
         if end_of_notes == -1:
-            end_of_notes = cls_docstring.find('        Examples\n')
+            end_of_notes = cls_docstring.find("        Examples\n")
             if end_of_notes == -1:
                 end_of_notes = len(cls_docstring)
-        func.__doc__ = (cls_docstring[:start_of_notes + len(notes_header)] +
-                        notes +
-                        cls_docstring[end_of_notes:])
+        func.__doc__ = (
+            cls_docstring[: start_of_notes + len(notes_header)]
+            + notes
+            + cls_docstring[end_of_notes:]
+        )
         return func
+
     return _doc
 
 
-def indentcount_lines(lines):
-    ''' Minimum indent for all lines in line list
+def indentcount_lines(lines: Iterable[str]) -> int:
```

Comment:
I can't think of a better way to annotate this, and maybe you alreayd know this; but since `str` itself is also an `Iterable[str]`, this also accepts `lines: str` as argument 🤷🏻

Reply:
I actually didn't know that or think of that but yes that makes sense that str is Iterable[str] as well. If so I'm not sure of a way to type annotate that we want something that iterates over str but it can't be str without doing a runtime check which we shouldn't do.

Reply:
For sequences it's also the same story; I've seen some people use `list[str] | tuple[str, ...]` in such cases.
But personally I'd rather have parameter type annotations that are too broad instead of too narrow 🤷🏻

Reply:
Anyway, I just wanted to point this out I guess; there's no need to change this as far as I'm concerned.